### PR TITLE
whoami: init at 1.10.3

### PR DESCRIPTION
**Pull request by dvcorreia**
Tiny Go webserver that prints OS information and HTTP request to output.
I find it useful in some situations, like debug and demo purposes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/wh/whoami/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  vendorHash = "sha256-qDfkYIAymkFUtbKka9OLoYjT+S9KhOra2UtOvhoz5Mw=";
+
+  ldflags = [ "-s" ];
+
+  CGO_ENABLED = 0;
+
+  doCheck = false;
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/by-name/wh/whoami/package.nix:26]: Remove `with lib;`.

We try to avoid as much as possible the use of `with` in front of an attribute set.

**Review by drupol — CHANGES_REQUESTED** [review]
Hi!

Thanks for your PR.

The diff LGTM, except a minor nit. Also, it would be nice to split your PR in 2 commits.
One commit to add you as maintainer, one for adding `whoami`.

Thank you!

**Comment by dvcorreia** [comment]
Thanks a lot for the review, @drupol !

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Thank you for your first contribution !

`whoami` will appear in `search.nixos.org` in `unstable` in 2~3 days, you can follow that using: https://nixpkgs-tracker.ocfox.me/?pr=348647 Once it is in `nixos-unstable`, it will be published on `search.nixos.org`.

Keep PRs coming ;)

**Pull request merged**